multi: add deadline for first connection of new LNC conn

ellemouton opened the pull request:
In this PR, we add a deadline for the initial connection of an LNC
connection. So with this, the user is forced to use their pairing phrase
within a certain time frame. After this initial connection, future
connections are made with the second handshake version meaning that the
pairing phrase is rendered useless. By adding a time limit to the time
in which a user can use their pairing phrase, we reduce the risk created
by the users pairing phrase being leaked. The default time limit is set
to 10 minutes but can be customsed with the new `firstlncconndeadline`
flag.

The first commit in this PR adds a new `CreatedAt` field to `Session` so that 
we can have an absolute deadline.

Depends on https://github.com/lightninglabs/lightning-node-connect/pull/53

Fixes https://github.com/lightninglabs/lightning-terminal/issues/406

jamaljsr reviewed (approved):
tACK LGTM.. amazing turn around time 🥇

guggero commented on session/tlv.go:
```diff
@@ -24,6 +24,7 @@ const (
 	typeLocalPrivateKey tlv.Type = 10
 	typeRemotePublicKey tlv.Type = 11
 	typeMacaroonRecipe  tlv.Type = 12
+	typeCreatedAt       tlv.Type = 13
```
Yayy, TLV for the win :star_struck:

guggero commented on session_rpcserver.go:
```diff
@@ -214,15 +215,92 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
+
+			err = s.db.RevokeSession(pubKey)
+			if err != nil {
+				log.Debugf("error revoking session: "+"%v", err)
```
Any reason for not returning the error here? Or was this intended to be a `return fmt.Errorf()`? Also, string concatenation artifacts.

ellemouton replied:
erg yeah sorry 🙈 also, there should be a `return` after this 🙈

guggero commented on config.go:
```diff
@@ -302,21 +304,22 @@ func defaultConfig() *Config {
+		Loop:                 &loopDefaultConfig,
+		PoolMode:             defaultPoolMode,
+		Pool:                 &poolDefaultConfig,
+		FirstLNCConnDeadline: 10 * time.Minute,
```
nit: extract into constant?

guggero commented on session_rpcserver.go:
```diff
@@ -214,15 +215,92 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
+		}
+
+		// Start the deadline timer.
+		firstConnectionDeadline = time.NewTimer(
```
I think we could use `time.AfterFunc()` here that closes the `firstConnTimeout`. Then in the `onNewStatus` callback we only stop that timer and we don't need a whole other goroutine (the debug comment could be moved into the `onNewStatus` callback).

ellemouton replied:
omg yeeeesss!!! thank you :) updated!

guggero reviewed (commented):
Nice, looks pretty good and should be a great security improvement!
I left a suggestion for making the logic a bit easier to understand.

ellemouton reviewed (commented):
thanks @guggero ! updated as per you suggestion 🚀  much neater the way you have suggested :)

ellemouton commented:
hmmm the itest keeps failing....i must have broken something on the latest update.  looking into it

ellemouton commented:
hmmmm having the same flake on master actually.....

ellemouton commented:
hahaha found the issue 🙈 when I extracted the timeout into a variable, I used seconds instead of minutes 🙈  should be g2g now! I also added another commit that adds LNC client logging to the itests just to make it easier to debug these things in future

guggero reviewed (approved):
utACK, LGTM :tada:

jamaljsr reviewed (approved):
tACK 🔥

The deadline works as advertised. Thanks for this great update.
